lib/imagec/imagec.go

```diff
@@ -0,0 +1,651 @@
+// Copyright 2016 VMware, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package imagec
+
+import (
+	"crypto/sha256"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/url"
+	"os"
+	"path"
+	"runtime/debug"
+	"strings"
+	"sync"
+	"time"
+
+	"golang.org/x/net/context"
+
+	log "github.com/Sirupsen/logrus"
+
+	docker "github.com/docker/docker/image"
+	dockerLayer "github.com/docker/docker/layer"
+	"github.com/docker/docker/pkg/ioutils"
+	"github.com/docker/docker/pkg/progress"
+	"github.com/docker/docker/pkg/streamformatter"
+	"github.com/docker/docker/pkg/stringid"
+	"github.com/docker/docker/reference"
+
+	"github.com/vmware/vic/lib/apiservers/portlayer/models"
+	"github.com/vmware/vic/lib/guest"
+	"github.com/vmware/vic/lib/metadata"
+)
+
+var (
+	options = &Options{}
+	// https://raw.githubusercontent.com/docker/docker/master/distribution/pull_v2.go
+	sf = streamformatter.NewJSONStreamFormatter()
+	po progress.Output
+)
+
+// ImageCOptions wraps the cli arguments
+type Options struct {
+	Reference string
+
+	Registry string
+	Image    string
+	Tag      string
+
+	Destination string
+
+	Host string
+
+	Logfile string
+
+	Username string
+	Password string
+
+	Token *Token
+
+	Timeout time.Duration
+
+	Stdout bool
+	Debug  bool
+
+	Outstream io.Writer
+
+	InsecureSkipVerify bool
+	InsecureAllowHTTP  bool
+
+	Standalone bool
```

Comment:
I thought about removing it, but if someone is using the binary they should still have that option.